src/utils/userManagement.js

```diff
@@ -5,30 +5,47 @@ const config = require('../config');
 const logger = require('../logger').getLogger(module);
 const ManagementClient = require('auth0').ManagementClient;
 const request = require('request-promise');
+const redis = require('redis');
+const bluebird = require('bluebird'); bluebird.promisifyAll(redis.RedisClient.prototype);
+const ONE_HOUR = 60 * 60;
+const ONE_DAY = ONE_HOUR * 24;
 
 // TODO: Implement user details caching.
 function getUserDetails(uuid) {
-  return getApiToken()
-    .then(token => {
-      return new ManagementClient({
-        token: token,
-        domain: config.get('AUTH0_DOMAIN')
-      });
-    })
-    .then(auth0 => {
-      return auth0.getUsers({
-        fields: 'name,email,user_metadata,app_metadata', // User details field names to get from API.
-        q: 'app_metadata.dorbel_user_id: ' + uuid // Query to get users by app metadata dorbel user id.
-      });
-    })
-    .then(user => {
-      logger.debug(user);
-      return user;
+  const cache = redis.createClient(6379, config.get('REDIS_HOST'));
+  const cacheKeyName = 'auth0_users';
+
+  return cache.hgetAsync(cacheKeyName, uuid)
+    .then(result => {
+      if (result) {
+        return JSON.parse(result);
+      } else {
+        return getApiToken()
+          .then(token => {
+            return new ManagementClient({
+              token: token,
+              domain: config.get('AUTH0_DOMAIN')
+            });
+          })
+          .then(auth0 => {
+            return auth0.getUsers({
+              fields: 'name,email,user_metadata,app_metadata', // User details field names to get from API.
+              q: 'app_metadata.dorbel_user_id: ' + uuid // Query to get users by app metadata dorbel user id.
+            });
+          })
+          .then(user => {
+            cache.hset(cacheKeyName, uuid, JSON.stringify(user));
+            logger.debug(user);
```

Comment:
please always include message in logs.
This could be a good place to log for 'cache miss'